src/shared/machine-id-setup.c

```diff
@@ -27,22 +27,14 @@
 #include "virt.h"
 
 static int generate_machine_id(const char *root, sd_id128_t *ret) {
-        const char *dbus_machine_id;
-        _cleanup_close_ int fd = -EBADF;
         int r;
 
         assert(ret);
 
         /* First, try reading the D-Bus machine id, unless it is a symlink */
-        dbus_machine_id = prefix_roota(root, "/var/lib/dbus/machine-id");
-        fd = open(dbus_machine_id, O_RDONLY|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW);
-        if (fd >= 0) {
-                if (id128_read_fd(fd, ID128_FORMAT_PLAIN, ret) >= 0) {
-                        log_info("Initializing machine ID from D-Bus machine ID.");
-                        return 0;
-                }
-
-                fd = safe_close(fd);
+        if (id128_read(root, "/var/lib/dbus/machine-id", ID128_FORMAT_PLAIN, ret) >= 0) {
```

Comment:
This will follow symlinks, which we shouldn't do here. I'd do a regular `chase_and_open()` with CHASE_NOFOLLOW here and then use `id128_read_fd()`